easyapplybot.py

```diff
@@ -290,55 +335,160 @@ def is_present(button_locator):
+												log.info("Clicked the radio button %s", yes_locator)
+
+											#TODO Issue where if there are multiple lines that ask for number of years experience then years experience will be written twice
+											#TODO Need to add a configuration file with all the answer for these questions versus having them hardcoded.
+											#Some questions are asking how many years of experience you have in a specific skill
+											#Automatically put the number of years that I have worked.
+											if "How many years" in text and "experience" in text:
+												textField = testLabelElement.find_element(By.XPATH, textInput_locator[1])
+												time.sleep(1)
+												log.info("Attempting to click the text field for %s", textInput_locator)
+												self.browser.execute_script("arguments[0].click()", textField)
+												log.info("Clicked the text field %s", textInput_locator)
+												time.sleep(1)
+												log.info("Attempting to send keys to the text field %s", textInput_locator)
+												textField.send_keys("10")
+												log.info("Sent keys to the text field %s", textInput_locator)
+
+											#This should be updated to match the language you speak.
+											if "Do you" in text and "speak" in text:
+												if "English" in text:
+													yesRadio = testLabelElement.find_element(By.XPATH, yes_locator[1])
+													time.sleep(1)
+													log.info("Attempting to click the radio button for %s", yes_locator)
+													self.browser.execute_script("arguments[0].click()", yesRadio)
+													log.info("Clicked the radio button %s", yes_locator)
+												#if not english then say no.
+												else:
+													noRadio = testLabelElement.find_element(By.XPATH, no_locator[1])
+													time.sleep(1)
+													log.info("Attempting to click the radio button for %s", no_locator)
+													self.browser.execute_script("arguments[0].click()", noRadio)
+													log.info("Clicked the radio button %s", no_locator)
+
+
+
+										except Exception as e:
+											log.exception("Could not answer additional questions: %s", e)
+											log.error("Unable to submit due to error with no solution")
+											return submitted
+									attemptQuestions = False
+									log.info("no longer going to try and answer questions, since we have now tried")
+								else:
+									log.error("Unable to submit due to error with no solution")
+									return submitted
+
+
 					if button:
-						button.click()
-						time.sleep(random.uniform(1.5, 2.5))
-						if i in (2, 3):
-							submitted = True
-						break
-				if button == None:
-					print("Could not complete submission")
-					break
-				elif submitted:
-					print("Application Submitted")
-					break
-
-			time.sleep(random.uniform(1.5, 2.5))
-
-			#After submiting the application, a dialog shows up, we need to close this dialog
+						if button_locator == upload_locator:
```

Comment:
Can we talk about if this is necessary?

Reply:
See comment above.

We can create a feature ticket/issue/request and use it to track this slight redundancy that we have accidentally created.

:)